src/Post/New/Update.elm

```diff
@@ -0,0 +1,53 @@
+module Post.New.Update exposing (..)
+
+
+import Post.New.Messages exposing (..)
+import Post.Models exposing (Post)
+import Post.New.Commands exposing (addPost)
+
+
+import Routing.Utils exposing (reverse, navigationCmd)
+import Routing.Models exposing (..)
+
+
+import Post.Commands exposing (getPosts)
+
+
+update : Msg -> Post -> (Post, Cmd Msg)
+update msg model =
+  case msg of
+    SavePost posts post ->
+      (model, (addPost posts post))
+    AddPostSucceed posts ->
+      let
+        path =
+          reverse PostsRoute
+
+        navCmd =
+          navigationCmd path
+
+        lastPost =
+          List.head (List.reverse posts)
+
+        newPost =
+          case lastPost of
+            Just new -> new
+            otherwise -> model
+
+      in  
+        (newPost, navCmd)
+
+    AddPostFail _ ->
+      (model, Cmd.none)
```

Comment:
I don't know how much error handling we can do for now, but we could print a message, log or something like that for now.